internal/pkg/service/stream/storage/node/coordinator/fileimport/operator_test.go

```diff
@@ -0,0 +1,321 @@
+	branch := test.NewBranch(branchKey)
+	sourceKey := key.SourceKey{BranchKey: branchKey, SourceID: "my-source"}
+	source := test.NewHTTPSource(sourceKey)
+	ts.sink = dummy.NewSinkWithLocalStorage(key.SinkKey{SourceKey: source.SourceKey, SinkID: "my-sink"})
+	require.NoError(t, ts.dependencies.DefinitionRepository().Branch().Create(&branch, ts.clk.Now(), test.ByUser()).Do(ctx).Err())
+	require.NoError(t, ts.dependencies.DefinitionRepository().Source().Create(&source, ts.clk.Now(), test.ByUser(), "create").Do(ctx).Err())
+	require.NoError(t, ts.dependencies.DefinitionRepository().Sink().Create(&ts.sink, ts.clk.Now(), test.ByUser(), "create").Do(ctx).Err())
+}
+
+func (ts *testState) prepareFile(t *testing.T, ctx context.Context) model.File {
+	t.Helper()
+	files, err := ts.dependencies.StorageRepository().File().ListIn(ts.sink.SinkKey).Do(ctx).All()
+	require.NoError(t, err)
+	require.Len(t, files, 1)
+	require.Equal(t, model.FileWriting, files[0].State)
+	return files[0]
+}
+
+func (ts *testState) prepareSlice(t *testing.T, ctx context.Context) model.Slice {
+	t.Helper()
+	slices, err := ts.dependencies.StorageRepository().Slice().ListIn(ts.sink.SinkKey).Do(ctx).All()
+	require.NoError(t, err)
+	require.Len(t, slices, 1)
+	require.Equal(t, model.SliceWriting, slices[0].State)
+	return slices[0]
+}
+
+func TestFileImport(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	ts := setup(t, ctx)
+	defer ts.teardown(t)
+	ts.prepareFixtures(t, ctx)
+	file := ts.prepareFile(t, ctx)
+	slice := ts.prepareSlice(t, ctx)
+
+	ts.clk.Add(time.Second)
+	require.NoError(t, ts.dependencies.StorageRepository().File().Rotate(ts.sink.SinkKey, ts.clk.Now()).Do(ctx).Err())
+	require.NoError(t, ts.dependencies.StorageRepository().Slice().SwitchToUploading(slice.SliceKey, ts.clk.Now()).Do(ctx).Err())
+	require.NoError(t, ts.dependencies.StorageRepository().Slice().SwitchToUploaded(slice.SliceKey, ts.clk.Now(), false).Do(ctx).Err())
+	ts.logger.Truncate()
+	require.NoError(t, ts.dependencies.StorageRepository().File().SwitchToImporting(file.FileKey, ts.clk.Now(), false).Do(ctx).Err())
+
+	// Check state
+	files, err := ts.dependencies.StorageRepository().File().ListIn(ts.sink.SinkKey).Do(ctx).All()
+	require.NoError(t, err)
+	assert.Len(t, files, 2)
+	assert.Equal(t, model.FileImporting, files[0].State)
+	assert.Equal(t, model.FileWriting, files[1].State)
+	slices, err := ts.dependencies.StorageRepository().Slice().ListIn(ts.sink.SinkKey).Do(ctx).All()
+	require.NoError(t, err)
+	assert.Len(t, slices, 2)
+	assert.Equal(t, model.SliceUploaded, slices[0].State)
+	assert.Equal(t, model.SliceWriting, slices[1].State)
+
+	ts.clk.Add(ts.interval)
+
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		ts.logger.AssertJSONMessages(c, `{"level":"info","message":"successfully imported file","component":"storage.node.operator.file.import"}`)
+	}, 5*time.Second, 10*time.Millisecond)
+
```

Comment:
Isn't it in the wrong order?

First, I wait until the mirror syncs to the new state, then I trigger `checkFile` via `clk.Add`.

Reply:
In addition, some comments would be good, since it is not entirely clear why it is there - if you don't know - in detail - how it works.